### PR TITLE
Test with bundled prover instead of local prover.

### PR DESCRIPTION
**Pull request by nuttycom**

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -12,21 +12,6 @@ jobs:
 
     steps:
       - uses: actions/checkout@v4
-      - name: Fetch path to Zcash parameters
-        working-directory: ./zcash_proofs
-        shell: bash
-        run: echo "ZCASH_PARAMS=$(cargo run --release --example get-params-path --features directories)" >> $GITHUB_ENV
```
Comment (str4d) [inline .github/workflows/ci.yml:18]: We can delete the `get-params-path` example now; it was only added for use in our CI. (However, we should verify that zebrad's CI isn't also using it; I think they might have been at one stage.)

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -12,21 +12,6 @@ jobs:
...
-      - name: Cache Zcash parameters
-        id: cache-params
-        uses: actions/cache@v3.3.2
-        with:
-          path: ${{ env.ZCASH_PARAMS }}
-          key: ${{ runner.os }}-params
-      - name: Fetch Zcash parameters
-        if: steps.cache-params.outputs.cache-hit != 'true'
-        working-directory: ./zcash_proofs
-        run: cargo run --release --example download-params --features download-params
```
Comment (str4d) [inline .github/workflows/ci.yml:28]: We can delete the `download-params` example now; it was only added for use in our CI. (However, we should verify that zebrad's CI isn't also using it; I think they might have been at one stage.)

**Review by str4d — APPROVED** [review]
ACK 3cdb0054f7f8ef1bf873cfcd2d4e7e113619d1fb

File: .github/workflows/ci.yml (hunk 1 above)
Comment (nuttycom) [reply .github/workflows/ci.yml:18]: `git grep` shows no usage of `get-params-path` in the zebrad repo.

File: .github/workflows/ci.yml (hunk 2 above)
Comment (nuttycom) [reply .github/workflows/ci.yml:28]: git grep shows no usage of download-params in the zebrad repo.

File: .github/workflows/ci.yml (hunk 1 above)
Comment (str4d) [reply .github/workflows/ci.yml:18]: Aha, what they actually did is copied those files into their CI. They removed their copies in https://github.com/ZcashFoundation/zebra/pull/7800.

Comment (str4d) [reply .github/workflows/ci.yml:18]: So yes, we can just remove `zcash_proofs/examples/` entirely (I don't think we need to offer Sprout parameter downloading examples either).

Comment (nuttycom) [reply .github/workflows/ci.yml:18]: Removed.

**Review by str4d — APPROVED** [review]
utACK c0a0ef1653e4806ae1e3475e10c1ede0ebaa3992.

**Pull request merged**